azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/impl/AppServicePlan.java

```diff
@@ -30,36 +36,132 @@ public AppServicePlan(AppServicePlanEntity appServicePlanEntity, AzureAppService
 
     @Override
     public IAppServicePlanCreator create() {
-        return null;
-    }
-
-    @Override
-    public IAppServicePlanUpdater update() {
-        return null;
+        return new AppServicePlanCreator();
     }
 
     @Override
     public boolean exists() {
-        return false;
+        return getPlanService(true) != null;
     }
 
     @Override
     public AppServicePlanEntity entity() {
-        return null;
+        return entity;
     }
 
     @Override
     public List<IWebApp> webapps() {
-        return null;
+        return getPlanService().manager().webApps().list().stream()
+                .filter(webapp -> StringUtils.equals(webapp.appServicePlanId(), getPlanService().id()))
+                .map(webapp -> new WebApp(AppServiceUtils.getBasicWebAppEntity(webapp), azureAppService))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public AppServicePlanUpdater update() {
+        return new AppServicePlanUpdater();
+    }
+
+    public com.azure.resourcemanager.appservice.models.AppServicePlan getPlanService() {
+        return getPlanService(false);
+    }
+
+    public synchronized com.azure.resourcemanager.appservice.models.AppServicePlan getPlanService(boolean force) {
+        if (appServicePlanClient == null || force) {
+            appServicePlanClient = AppServiceUtils.getAppServicePlan(entity, azureClient);
+        }
+        return appServicePlanClient;
     }
 
     @Override
     public String id() {
-        return null;
+        return getPlanService().id();
     }
 
     @Override
     public String name() {
-        return null;
+        return getPlanService().name();
+    }
+
+    public class AppServicePlanCreator implements IAppServicePlanCreator {
+        private String name;
+        private Region region;
```

Comment:
shall we share the common fields in one base class: the name region, resource group?

Reply:
only name is shared by resources, for instance, region and resource group could not be set for `DeploymentSlotCreator`